synergy-design-system/synergy-design-system
# feat: ✨ Styles Package

# Pull Request

This PR adds our new styles package as originally defined in #365. It adds the following functionality:

## 📖 Description

## 1. `@synergy-design-system/styles`

A new package named `@synergy-design-system/styles` was added. It currently holds a `typography` module only. Please have a look at the provided `README.md` to see how it is **supposed** to work.

## 2. Package Updates

### Demos:

- Adjusted demos so that they make use of the styles package.
- Added some samples on the main page of each of them.

### Wrappers:

- Adjusted each README to take into account the new styles package

### Components:

- Adjusted each README to take into account the new styles package
- Added `@synergy-design-system/tokens` as a `peerDependency`
- Added ability to vendor css files, too (**this means all files vendored where updated!**)
- Added new namespace `styles/*` to `exports` map in package.json to make sure styles can be loaded.
- Added `_utility.css` from shoelace to vendored files (will get moved to `src/styles/utility.css`)
- Added `fouc.css` that adds scoped `fouc` support.
- Added generation of new style utilities to the build process to make creation of styles dynamic.

### Storybook:

- Adjusted storybook configuration that makes use of the styles package 
- Added README and CHANGELOG of the new styles package
- Adjusted configuration to add the styles to the parsed manifest file to make sure knobs work as intended.
- A new story namespace `Styles` was created, holding two demo stories.
- Added a demo that shows usage of `.syn-body` helper classes.

### 🎫 Issues

- Closes #455
- Closes #436 
- Closes #269



## 👩‍💻 Reviewer Notes

The following changes where suggested and are not **yet** part of the package:

- [x] Build does not run as my custom markdown is not formatted correctly for P R e T t I e R :(
- [x] Allow the usage of automatic boolean attributes for classes
- [x] Use BEM-Syntax for classes
- [x] Currently, the Manifest does not reload when one of the files in the styles package is updated.

## 📑 Test Plan

- Check out this branch and run `pnpm i -r && pnpm build`.
- Play around with the new package and create a new module by following the supplied `README.md`
- Run the build process and see if the `README.md` gets updated.
- Launch Storybook and have a look at the new namespace `Styles`
- Change a value in one of the style modules and reload storybook. Notice how the stories "className" property will get updated to include the new className (or have one item removed if you chose to remove one)

## ✅ DoD
